nixos/modules/misc/ids.nix

```diff
@@ -275,6 +275,8 @@
       gocd-server = 252;
       terraria = 253;
       mattermost = 254;
+      quagga = 255;
```

Comment:
There is no specific reason to use static ids, I just assumed it to be good practice for daemons, according to the comment at the top of misc/ids.nix.

I've updated the module to use dynamic ids, and rebased the PR.